dandiapi/api/management/commands/extract_metadata.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import annotations
+
+import logging
+from pathlib import Path
+
+from dandi.dandiapi import RemoteReadableAsset
+from dandi.metadata.nwb import nwb2asset
+from dandischema.models import get_schema_version
+from django.contrib.auth.models import User
+from django.db import transaction
+import djclick as click
+from tqdm import tqdm
+
+from dandiapi.api.models import Asset, Dandiset, Version
+from dandiapi.api.services.asset import change_asset
+
+logger = logging.getLogger(__name__)
+
+
+@click.group()
+def group():
+    pass
+
+
+def extract_asset_metadata(asset: Asset, draft_version: Version):
+    readable_asset = RemoteReadableAsset(
+        asset.s3_url, size=asset.size, mtime=asset.modified, name=Path(asset.path).name
+    )
+
+    if not asset.path.lower().endswith('.nwb'):
+        logger.info('Asset %s: Not an NWB file, skipping...', asset.path)
+        return
+
+    new_metadata = nwb2asset(readable_asset, schema_version=get_schema_version()).json_dict()
```

Comment:
if would be close to what we use (for nwb only) in [NWBAsset.get_metadata](https://github.com/dandi/dandi-cli/blob/HEAD/dandi/files/bases.py#L475)

notable differences:
- we provide digest (not schema_version) . I guess should be added here too?
- we fallback to default record if extraction fails for some reason
- we assign .path explicitly to be posix path from the top of the dandiset, but I think here it would be matching fine the one of RemoteReadableAsset (worth checking on example)

Reply:
I've updated my script to provide the digest as well. I chose not to automatically handle errors as I'm not sure that's the behavior we want. Since this is a management command to be run manually, it's alright to manually intervene in that failure mode.